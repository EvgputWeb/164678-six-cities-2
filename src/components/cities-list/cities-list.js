import React from 'react';
import PropTypes from 'prop-types';

const CitiesList = ({list, selectedCity, onCityClick}) => {
  return (
    <div className="tabs">
      <section className="locations container">
        <ul className="locations__list tabs__list">

          {list.map((city) => (
            <li key={city} className="locations__item">
              <a href="#"
                className={`locations__item-link tabs__item` + ((city === selectedCity) && (` tabs__item--active`)) }
                onClick={(e)=>{
                  e.preventDefault();
                  onCityClick(city);
                }}
              >
                <span>{city}</span>
              </a>
            </li>
          ))}

        </ul>
      </section>
    </div>
  );
};


CitiesList.propTypes = {
  list: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
  selectedCity: PropTypes.string.isRequired,
  onCityClick: PropTypes.func.isRequired
};


export default CitiesList;
