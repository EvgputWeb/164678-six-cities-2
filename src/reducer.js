const getOffersByCity = (allOffers, cityName) => allOffers.filter((item) => item.city.name === cityName);

const initialState = {
  allOffers: [],
  city: ``,
  cityOffers: []
};


const ACTIONS = {
  LOAD_ALL_OFFERS: `LOAD_ALL_OFFERS`,
  GET_CITY_OFFERS: `GET_CITY_OFFERS`,
  CHANGE_CITY: `CHANGE_CITY`,
};

const ActionCreator = {
  loadAllOffers: (offers) => {
    return {
      type: ACTIONS.LOAD_ALL_OFFERS,
      payload: offers
    };
  },
  changeCity: (city) => {
    return {
      type: ACTIONS.CHANGE_CITY,
      payload: city
    };
  },
  getCityOffers: (city) => {
    return {
      type: ACTIONS.GET_CITY_OFFERS,
      payload: city
    };
  }
};


const reducer = (state = initialState, action) => {
  switch (action.type) {

    case ACTIONS.LOAD_ALL_OFFERS: return Object.assign({}, state, {
      allOffers: action.payload
    });

    case ACTIONS.CHANGE_CITY: return Object.assign({}, state, {
      city: action.payload
    });

    case ACTIONS.GET_CITY_OFFERS: return Object.assign({}, state, {
      cityOffers: getOffersByCity(state.allOffers, action.payload)
    });
  }

  return state;
};


const Operation = {
  loadAllOffers: () => (dispatch, _, api) => {
    return api.get(`/hotels`)
      .then((response) => {
        dispatch(ActionCreator.loadAllOffers(response.data));
        const initialCity = response.data[0].city.name;
        dispatch(ActionCreator.changeCity(initialCity));
        dispatch(ActionCreator.getCityOffers(initialCity));
      });
  }
};


export {ActionCreator, reducer, Operation};

export {initialState, ACTIONS, getOffersByCity}; // использую в  reducer.test.js и в app.test.js
