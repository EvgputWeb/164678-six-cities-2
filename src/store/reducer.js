import {combineReducers} from 'redux';
import {allOffers} from './reducers/load-data';
import {city} from './reducers/change-city';

const reducer = combineReducers({
  allOffers,
  city
});

export {reducer};
