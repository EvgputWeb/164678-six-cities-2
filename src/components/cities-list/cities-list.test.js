import React from 'react';
import renderer from 'react-test-renderer';
import {CitiesList} from './cities-list';
import {initialState} from '../../reducer';


it(`CitiesList renders correctly`, () => {
  const citiesSet = new Set(initialState.allOffers.map((offer) => offer.city.name));
  const list = [...citiesSet].slice(0, 5);

  const citiesList = renderer
    .create(
        <CitiesList
          list={list}
          activeItem={initialState.city}
          onActivateItem={jest.fn()}
          cityClickAction={jest.fn()}
        />
    )
    .toJSON();

  expect(citiesList).toMatchSnapshot();
});
