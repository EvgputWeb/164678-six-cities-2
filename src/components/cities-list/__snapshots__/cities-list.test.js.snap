// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CitiesList renders correctly 1`] = `
<div
  className="tabs"
>
  <section
    className="locations container"
  >
    <ul
      className="locations__list tabs__list"
    />
  </section>
</div>
`;
